docs/2.guide/1.concepts/10.nuxt-lifecycle.md

```diff
@@ -0,0 +1,135 @@
+::
+
+:read-more{to="/docs/guide/directory-structure/server#server-middleware"}
+
+### Step 3: Initialize Nuxt and Execute Nuxt Server Plugins
+
+The Vue and Nuxt instances are created first. Afterward, Nuxt executes its server plugins. This includes:
+- Built-in plugins, such as Vue Router and `unhead`.
+- Custom plugins located in the `plugins/` directory, including those without a suffix (e.g., `myPlugin.ts`) and those with the `.server` suffix (e.g., `myServerPlugin.server.ts`).
+
+Plugins execute in a specific order and may have dependencies on one another. For more details, including execution order and parallelism, refer to the [Plugins documentation](/docs/guide/directory-structure/plugins).
+
+::callout{icon="i-ph-lightbulb"}
+After this step, Nuxt calls the [`app:created`](/docs/api/advanced/hooks#app-hooks-runtime) hook, which can be used to execute additional logic.
+::
+
+:read-more{to="/docs/guide/directory-structure/plugins"}
+
+### Step 4: Route Validation
+
+After initializing plugins and before executing middleware, Nuxt calls the `validate` method if it is defined in the `definePageMeta` function. The `validate` method, which can be synchronous or asynchronous, is often used to validate dynamic route parameters.
+
+- The `validate` function should return `true` if the parameters are valid.
+- If validation fails, it should return `false` or an object containing a `statusCode` and/or `statusMessage` to terminate the request.
+
+For more information, see the [Route Validation documentation](/docs/getting-started/routing#route-validation).
+
+:read-more{to="/docs/getting-started/routing#route-validation"}
+
+### Step 5: Execute Nuxt Server Middleware
+
+Middleware allows you to run code before navigating to a particular route. It is often used for tasks such as authentication, redirection, or logging.
+
+In Nuxt, there are three types of middleware:
+- **Global route middleware**
+- **Named route middleware**
+- **Anonymous (or inline) route middleware**
+
+Nuxt automatically executes global middleware for first time enter to the application and every time before route navigation. Named and anonymous middleware are executed only on the routes specified in the middleware property of the page(route) meta defined in the corresponding page components.
+
+For details about each type and examples, see the [Middleware documentation](/docs/guide/directory-structure/middleware).
+
+:read-more{to="/docs/guide/directory-structure/middleware"}
+
+### Step 6: Setup Page and Components
+
+Nuxt initializes the page and its components during this step and fetches any required data with `useFetch` and `useAsyncData`. Since there are no dynamic updates and no DOM operations occur on the server, Vue lifecycle hooks such as `onBeforeMount`, `onMounted`, and subsequent hooks are **NOT** executed during SSR.
+
```

Comment:
```suggestion
::important
You should avoid code that produces side effects that need cleanup in root scope of `<script setup>`. An example of such side effects is setting up timers with `setInterval`. In client-side only code we may setup a timer and then tear it down in `onBeforeUnmount` or `onUnmounted`. However, because the unmount hooks will never be called during SSR, the timers will stay around forever. To avoid this, move your side-effect code into `onMounted` instead.
::

```

Adding this section copied from [Vue docs](https://vuejs.org/guide/scaling-up/ssr.html#component-lifecycle-hooks) as very important to avoid memory leaks